### PR TITLE
Adplotzk patch 1

### PR DESCRIPTION
**Pull request by adplotzk**
Adding a checksum to ensure downloads are from valid AWS source.

### CODE REVIEW

File: validate_1.0.0-rc.2/checksum.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# !/usr/bin/env python3
+# -*- coding: utf-8 -*-
+ 
+ 
+import hashlib
+import os
+ 
+def get_checksum(filename, hash_function):
+    """Generate checksum for file baed on hash function (MD5 or SHA256).
```
Comment (floydtree) [inline validate_1.0.0-rc.2/checksum.py:9]: All these comments seem unnecessary, considering you have hardcoded algorithm and input in the code below. Can be removed

File: validate_1.0.0-rc.2/README.md [hunk 2]
```diff
@@ -66,7 +67,24 @@ The expected output from running the sample is as follows:
...
 		VALID OCSF.
-		
+
+## Checksum <a name="Checksum"></a>
+
+To ensure the soundness of the download, you may choose run a checksum against the files in this repository.
+
+In order to generate the expected hash and run the checksum please run the following:
+
+	tar -czvf hash.tar.gz /path/to/validate_1.0.0-rc.2/ setup.py validate.py sample requirements.txt sample README.md ocsf_schema_1.0.0-rc.2 docs LICENSE
```
Comment (floydtree) [inline validate_1.0.0-rc.2/README.md:77]: Is this tested? Also, adding in /path/to/validate_1.0.0-rc.2/ seems unnecessary and won't allow for direct copy-paste, instead why not have cd statement in the first line and make everything else copy-paste-able?

File: validate_1.0.0-rc.2/checksum/hash.tar.gz
Comment (floydtree) [inline validate_1.0.0-rc.2/checksum/hash.tar.gz:1]: why push this file?

Comment (floydtree) [reply validate_1.0.0-rc.2/checksum/hash.tar.gz:1]: plus I extracted it, seems like you included checksum.py in the tar but that's now present in the tar command you mentioned in the readme? This yields in different checksums.

File: validate_1.0.0-rc.2/checksum.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        bytes = f.read()  # read file as bytes
+        if hash_function == "sha256":
+            readable_hash = hashlib.sha256(bytes).hexdigest()
+        else:
+            Raise("{} is an invalid hash function. Please Enter MD5 or SHA256")
+ 
+    return readable_hash
+
+hashcode = "checksum/hash.tar.gz"
+hashtest = "462000e7b041f54b1b8e4ef59e4372f62f9bb2d293c3f9122a4f19c46edadf17"
```
Comment (floydtree) [inline validate_1.0.0-rc.2/checksum.py:34]: check my comment below, don't think this checksum would match considering your tar command in the readme

**Review by floydtree — CHANGES_REQUESTED** [review]

File: validate_1.0.0-rc.2/checksum.py (hunk 1 above)
Comment (adplotzk) [reply validate_1.0.0-rc.2/checksum.py]: this has been removed

File: validate_1.0.0-rc.2/README.md (hunk 2 above)
Comment (adplotzk) [reply validate_1.0.0-rc.2/README.md]: this has been tested, copy and paste commands have been implemented

File: validate_1.0.0-rc.2/checksum/hash.tar.gz
Comment (adplotzk) [reply validate_1.0.0-rc.2/checksum/hash.tar.gz:1]: this file was there for demonstrative purposes but has been removed along with an update to the script that will create this directory during runtime

File: validate_1.0.0-rc.2/checksum.py (hunk 3 above)
Comment (adplotzk) [reply validate_1.0.0-rc.2/checksum.py]: checksum now succeeds

**Pull request closed without merging**